File: Aplikacija/Backend/LabZakazivanjeAPI/Models/Session.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LabZakazivanjeAPI.Models;

public enum SessionState
{
    PLANNED = 0,
    NEXT = 1,
    ACTIVE = 2,
    FADING = 3,
    FINISHED = 4
};


public class Session
{
    [Key]
    public int Id {get; set;}
    public int RoomId {get; set;}
    public int ActivityId {get; set;}

    [ForeignKey(nameof(RoomId))]
    public Room? Prostorija {get; set;}

    [ForeignKey(nameof(ActivityId))]
    public Activity? Aktivnost {get; set;}
    
    public required DateOnly Datum {get; set;}
    public required TimeOnly VremePocetka {get; set;}
    public required TimeOnly VremeKraja {get; set;}
    
    public required SessionState Stanje {get; set;}

    public required bool AutomatskiPocetak {get; set;}
    public required bool AutomatskiKraj {get; set;}
    public SessionState? AutomatskoStanjeZavrsavanja {get; set;}

    public override string ToString()
    {
        return $"{Aktivnost.Name} {Prostorija.Naziv} {Stanje}";
    }
}

File: Aplikacija/Frontend/lab_zakazivanje/src/DTOs/session.js
export class SessionView
{
    static Statuses = Object.freeze({
        0:"planned",
        1:"next",
        2:"active",
        3:"fading",
        4:"finished"
    })

    constructor(id, naziv, tip, datum, vremePoc, vremeKraja, status, autoStart, autoEnd, autoState)
    {
        this.id = id;
        this.naziv = naziv;
        this.tip = tip;
        this.datum = datum;
        this.vremePoc = vremePoc;
        this.vremeKraja = vremeKraja;
        this.status = SessionView.Statuses[status];
        this.autoStart = autoStart;
        this.autoEnd = autoEnd;
        this.autoState = autoState;
    }
}